pycaching/geocaching.py

```diff
@@ -228,7 +228,7 @@ def logout(self):
         logging.info("Logging out.")
         self._logged_in = False
         self._logged_username = None
-        self._session = requests.Session()
+        self._session.cookies.clear()
```

Comment:
I assume that this should fix the failing logout test?

Reply:
Not only – it was some weird combination of improper logout handling (this), incomplete mocking (see `test/__init__.py` change), usage of `setUpClass` (thus keeping the polluted Geocaching object available between test cases), executing multiple subtests in a single test method (the same issue with polluted objects) and last but not least confusing cassette naming and separation.